File: CompetitiveProgramming/src/edu/algos/bit/DetectTwoNumbersWithOpositeSigns.java
package edu.algos.bit;

public class DetectTwoNumbersWithOpositeSigns {

	public static void main(String[] args) {
		int num1 = -1;
		int num2 = 100;
		System.out.println(detectTwoNumbersWithOpositeSigns(num1, num2));
	}

	// for negative numbers, the last bit is 1 which is sign bit
	// for positive nums it is 0
	// if we xor these 2 nums. result should be -ve if there are of opposite signs
	private static boolean detectTwoNumbersWithOpositeSigns(int num1, int num2) {
		return (num1^num2)<0;
	}
	
}

File: dsa/topics/19_bit/python/detect_two_numbers_with_oposite_signs.py
# For negative numbers, the last bit is 1 which is sign bit
# For positive nums it is 0
# If we xor these 2 nums. result should be -ve if there are of opposite signs
def detect_two_numbers_with_opposite_signs(num1, num2):
    return (num1 ^ num2) < 0


# ------------------ Test (Same as Java main) ------------------

num1 = -1
num2 = 100
print(detect_two_numbers_with_opposite_signs(num1, num2))  # Expected: True
